### ISSUE TITLE
oncomplete js lazy analysis EL

### ISSUE THREAD

**Issue by cooper-lyt** [TheCoder4eu/BootsFaces-OSP#410]
```xml
    <b:commandButton ajax="true" value="确定"
        oncomplete="if (#{facesContext.maximumSeverity==null} || #{facesContext.maximumSeverity.ordinal lt 2}){$('.modalEdit').modal('hide')}"
        action="#{officeHome.save}" look="success" update="modalInputs,resultData">
    </b:commandButton>
```

is java script worked in Richfaces , but not in bootsfaces!

I think it is lazy analysis EL,that is a good idea!

**Comment by zhedar** [comment]
I've formatted your code a bit, so it's fully readable.

I'm currently not sure, how this could work out. But I've also never used RichFaces anyway. I'm guessing the EL should be evaluated, after the Ajax request took place, but it is evaluated at page load time?

This looks like something, that can be done with Ajax though by using the `onsuccess` callback.

**Comment by cooper-lyt** [comment]
I'm guessing "oncomplete" evaluated is String at  page load time!  

 if has a faces error message  "onsuccess" is not run?

**Comment by zhedar** [comment]
According to the documentation, if there is a Java Exception, it will not be called. So maybe a `ValidatorException` will suffice, but I'm just guessing here. You can also [set a FacesMessage this way](http://stackoverflow.com/questions/10007438/how-can-i-display-validatorexception-and-required-true-of-same-input-field-in).

**Comment by cooper-lyt** [comment]
` <b:inputText id="name" value="#{officeHome.instance.name}" required="true"
                             label="name" renderLabel="true"/>
<h:message for="name" />`

` <b:commandButton ajax="true" value="确定"
                                     onsuccess="alert('onsuccess')"
                                        oncomplete="alert('oncomplete')"
                                     action="#{officeHome.save}" look="success" update="modalInputs,resultData">
  </b:commandButton>`

onsuccess still be execute!

need create a Validator?

**Comment by stephanrauh** [comment]
@cooper-lyt I agree with you that this is a nice feature. Actually, we've already got a similar feature. Have a look at http://www3.bootsfaces.net/Showcase/forms/FetchBeanInfos.jsf. We can add the severity bit if you like, but the basic functionality is already there.

**Comment by cooper-lyt** [comment]
@stephanrauh 
fetchBeanInfos not work for me
`      <h:form>
            <h:panelGrid id="inform" columns="3" cellpadding="5">
                <h:outputLabel for="firstname" value="Firstname: *" />
                <b:inputText id="firstname" value="#{officeHome.id}">
                    <f:validateLength minimum="2" />
                </b:inputText>
                <b:fetchBeanInfos />
            </h:panelGrid>
            <b:column span="2">
                <b:commandButton value="Ajax Modal" ajax="true" update="@form:inform" id="ajaxmodal" look="inverse"

                                 oncomplete="alert(validationFailed);" />
            </b:column>
 </h:form>`

always is "false",What did I miss?

**Comment by cooper-lyt** [comment]
i found it !  
fetchBeanInfos only work for bean annotation，only  `<f:validateLength` minimum="2" />` or  required will be not work!

**Comment by stephanrauh** [comment]
The `event.data` bit seems to be an interesting concept. Maybe we're going to add something like this to BootsFaces, too. Let me ponder about it for a while.

**Comment by stephanrauh** [comment]
Solved by pull request #430.
